src/Chirp.Infrastructure/ChirpRepository/AuthorRepository.cs

```diff
@@ -7,130 +7,161 @@
     protected ChirpDBContext context;
 
     public AuthorRepository(ChirpDBContext dbContext)
     {
-        DbSet = dbContext.Authors;
+        DbSetAuthor = dbContext.Authors;
         context = dbContext;
     }
 
     #region IAuthorRepository<Author, Cheep> Members
 
     public void Insert(Author entity)
     {
-        DbSet.Add(entity);
+        DbSetAuthor.Add(entity);
         context.SaveChanges();
     }
 
     public void Delete(Author entity)
     {
-        DbSet.Remove(entity);
+        DbSetAuthor.Remove(entity);
+        context.SaveChanges();
     }
 
     public IQueryable<Author> SearchFor(Expression<Func<Author, bool>> predicate)
     {
-        return DbSet.Where(predicate);
+        return DbSetAuthor.Where(predicate);
     }
 
     public async Task<Author> GetAuthorWithCheeps(string authorName)
     {
         // The returned Author object can use Author.Cheeps to get all the Cheeps (sorted by descending timestamp)
         
-        var author = await DbSet.Include(_author => _author.Cheeps
+        var author = await DbSetAuthor.Include(_author => _author.Cheeps
                     .OrderByDescending(_cheep => _cheep.TimeStamp))
-                    .Where(_author => _author.Name == authorName)
+                    .Where(_author => _author.User.Name == authorName)
                     .FirstOrDefaultAsync() ?? throw new Exception($"Author {authorName} not found");
         return author;
     }
 
+    public async Task<int> GetCheepsCountsFromAuthorId(int id) 
+    {
+        var authorEntity = SearchFor(_author => _author.User.UserId == id).FirstOrDefault();
+
+        if (authorEntity == null) {
+            return 0; // make this into an exception
+        }
+
+        int cheepsCount = DbSetAuthor.Entry(authorEntity)
+                    .Collection(_author => _author.Cheeps)
+                    .Query()
+                    .Count();
+        return cheepsCount;
+    }
+
+    public async Task<Author?> GetAuthorByName(string name)
+    {
+        // FirstOrDefault returns null if no User is found.
+        var author = await SearchFor(_author => _author.User.Name == name).FirstOrDefaultAsync();
+
+        return author;
+    }
+
+    //GetCheepsByAuthor should be replaced with GetCheepsByAuthorId, we should search by id not name
     public async Task<Tuple<List<CheepDTO>, int>> GetCheepsByAuthor(string author, int offset, int limit)
     {
         // Helge has said we're to assume Author.Name are unique for now.
 
         int cheepsCount = 0;
 
-        var authorEntity = await GetAuthorByName(author);
+        var authorEntity = SearchFor(_author => _author.User.Name == author).FirstOrDefault();
 
         if (authorEntity is null)
         {
             return new Tuple<List<CheepDTO>, int>(new List<CheepDTO>(), cheepsCount);
         }
         else
         {
-            cheepsCount = DbSet.Entry(authorEntity)
+            cheepsCount = DbSetAuthor.Entry(authorEntity)
```

Comment:
would rec calling `GetCheepsCountFromAuthorID(GetAuthorByAuthorId)` or making a method `GetCheepsCountFromAuthorObject` for cleaner code.

Reply:
It is a good idea, but it is out of scope for this branch